### PR TITLE
Promote CronJob TZ to GA

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind feature
/kind api-change
/sig apps

#### What this PR does / why we need it:
This promotes `CronJobTimeZone` feature gate to GA. 

#### Special notes for your reviewer:
/assign @ravisantoshgudimetla @atiratree 
for controller changes

#### Does this PR introduce a user-facing change?
```release-note
Promote CronJobTimeZone feature to GA
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/3140
```

### CODE REVIEW

**Comment by sftim** [comment]
BTW, https://github.com/kubernetes/enhancements/issues/3140#issuecomment-1082237075

> &hellip;[#3143 (comment)](https://github.com/kubernetes/enhancements/pull/3143#discussion_r837782004) strongly suggests to me that we should document how to manage a custom timezone database (via a `hostPath` volume mount or similar)&hellip;

I do recommend documenting that - maybe as a blog article that doesn't expire?

**Review by ravisantoshgudimetla — APPROVED** [review]
Thank you for working on this @soltysh 

/lgtm
/approve

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -888,7 +889,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	ConsistentHTTPGetHandlers: {Default: true, PreRelease: featuregate.GA},
 
-	CronJobTimeZone: {Default: true, PreRelease: featuregate.Beta},
+	CronJobTimeZone: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.28
```
Comment (pacoxu) [inline pkg/features/kube_features.go:892]: Nit: according to the skew policy, better `remove in 1.29` after two releases.

**Comment by liggitt** [comment]
API bits lgtm, comment on feature gate dropping in 1.29 is my only comment on the PR. are all the graduation bits complete? any tests that need updating or enabling by default, doc PRs, etc?

**Pull request merged**